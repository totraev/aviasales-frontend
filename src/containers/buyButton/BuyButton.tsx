import React, { SFC } from 'react';
import { connect } from 'react-redux';
import styles from './BuyButton.module.css';

import { currencySign, priceWithSpaces } from '../../helpers/currencies';

import { IState } from '../../redux/rootReducer';
import { IBuyButtonProps, IStateProps } from './BuyButton.types';

export const BuyButton: SFC<IBuyButtonProps> = ({ currency, currencyPrice, price }) => (
  <button className={styles.btn}>
    Купить
    <br/>
    за {priceWithSpaces(Math.round(price / currencyPrice))} ​{currencySign(currency)}
  </button>
);

export default connect<IStateProps, {}, {}, IState>(
  ({ filter: { currentCurr, currencyMap } }) => ({
    currency: currentCurr,
    currencyPrice: currencyMap[currentCurr],
  }),
)(BuyButton);
