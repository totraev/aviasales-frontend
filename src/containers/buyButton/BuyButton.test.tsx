import React from 'react'
import renderer from 'react-test-renderer'

import { BuyButton } from './BuyButton'

describe('BuyButton component', () => {
  it('Should renders correctly', () => {
    const tree = renderer
      .create(<BuyButton currency="rub" currencyPrice={1} price={500} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
