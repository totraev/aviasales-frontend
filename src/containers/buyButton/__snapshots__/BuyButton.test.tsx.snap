// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuyButton component Should renders correctly 1`] = `
<button
  className="btn"
>
  Купить
  <br />
  за  
  <span
    className="rub"
  >
    500
  </span>
</button>
`;
